client/overview/inbox-notifications/index.js

```diff
@@ -0,0 +1,315 @@
+
+	return (
+		<TransitionGroup role="menu">
+			{ notesArray.map( ( note ) => {
+				const { id: noteId, is_deleted: isDeleted } = note;
+				if ( isDeleted ) {
+					return null;
+				}
+				return (
+					<CSSTransition
+						key={ noteId }
+						timeout={ 500 }
+						classNames="wcpay-inbox-message"
+					>
+						<InboxNoteCard
+							key={ noteId }
+							note={ note }
+							lastRead={ lastRead }
+							onDismiss={ onDismiss }
+							onNoteActionClick={ onNoteActionClick }
+							onBodyLinkClick={ onBodyLinkClick }
+							onNoteVisible={ onNoteVisible }
+						/>
+					</CSSTransition>
+				);
+			} ) }
+		</TransitionGroup>
+	);
+};
+
+const InboxPanel = () => {
+	const { createNotice } = useDispatch( 'core/notices' );
+	const {
+		batchUpdateNotes,
+		removeAllNotes,
+		removeNote,
+		updateNote,
+		triggerNoteAction,
+	} = useDispatch( NOTES_STORE_NAME );
+	const {
+		isError,
+		resolving,
+		batchUpdating,
+		notes,
+		overviewInboxLastRead,
+	} = useSelect( ( select ) => {
+		const {
+			getNotes,
+			getNotesError,
+			isResolving,
+			isNotesRequesting,
+		} = select( NOTES_STORE_NAME );
+		const { getCurrentUser } = select( 'core' );
+		const currentUser = getCurrentUser();
+		const woocommerceMeta = currentUser ? currentUser.woocommerce_meta : {};
+
+		return {
+			notes: getNotes( INBOX_QUERY ),
+			isError: Boolean( getNotesError( 'getNotes', [ INBOX_QUERY ] ) ),
+			resolving: isResolving( 'getNotes', [ INBOX_QUERY ] ),
+			batchUpdating: isNotesRequesting( 'batchUpdateNotes' ),
+			overviewInboxLastRead: woocommerceMeta.wc_payments_overview_inbox_last_read
+				? JSON.parse(
+						woocommerceMeta.wc_payments_overview_inbox_last_read
+				  )
+				: undefined,
+		};
+	} );
+	const [ dismiss, setDismiss ] = useState();
+	// Make sure we only set lastRead at component mount.
+	const lastRead = useRef( overviewInboxLastRead );
+	if ( ! lastRead.current && overviewInboxLastRead ) {
+		lastRead.current = overviewInboxLastRead;
+	}
+
+	useEffect( () => {
+		const mountTime = Date.now();
+
+		const userDataFields = {
+			wc_payments_overview_inbox_last_read: mountTime,
+		};
+		updateWoocommerceUserMeta( userDataFields );
+	}, [] );
+
+	if ( isError ) {
+		const title = __(
+			'There was an error getting your inbox. Please try again.',
+			'woocommerce-admin'
+		);
+		const actionLabel = __( 'Reload', 'woocommerce-admin' );
+		const actionCallback = () => {
+			// @todo Add tracking for how often an error is displayed, and the reload action is clicked.
+			window.location.reload();
+		};
+
+		return (
+			<EmptyContent
+				title={ title }
+				actionLabel={ actionLabel }
+				actionURL={ null }
+				actionCallback={ actionCallback }
+			/>
+		);
+	}
+
+	const onDismiss = ( note, type ) => {
+		setDismiss( { note, type } );
+	};
+
+	const closeDismissModal = async ( confirmed = false ) => {
+		const noteNameDismissAll = 'all' === dismiss.type ? true : false;
```

Comment:
Nitpick: I think that this is the same as:
`const noteNameDismissAll = 'all' === dismiss.type;`

Reply:
Thanks, yeah that is nicer, updated in: https://github.com/Automattic/woocommerce-payments/pull/2053/commits/c9bc8d4413d9618c80a8439d086ca3de6ece02ea